watchers/chaos_pod_test.go

```diff
@@ -0,0 +1,181 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package watchers_test
+
+import (
+	"fmt"
+
+	"github.com/DataDog/chaos-controller/api/v1beta1"
+	"github.com/DataDog/chaos-controller/watchers"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+)
+
+var _ = Describe("Chaos Pod watcher", func() {
+	Describe("Handler", func() {
+		const (
+			bufferSize = 10
+		)
+
+		var (
+			chaosPodHandler watchers.ChaosPodHandler
+			disruption      *v1beta1.Disruption
+			eventRecorder   record.EventRecorder
+		)
+
+		JustBeforeEach(func() {
```

Comment:
Prefer `BeforeEach` whenever possible, in particular here it's configuration/setup, not a creation.

Use `JustBeforeEach` probably only once to perform the creation and associated assertions.

https://onsi.github.io/ginkgo/#separating-creation-and-configuration-justbeforeeach